savant/deepstream/encoding.py

```diff
@@ -0,0 +1,78 @@
+import time
+from typing import Any, Dict
+
+import pyds
+
+from savant.config.schema import PipelineElement
+from savant.gstreamer import Gst  # noqa:F401
+from savant.gstreamer.codecs import CODEC_BY_NAME, Codec
+from savant.gstreamer.element_factory import GstElementFactory
+from savant.gstreamer.runner import GstPipelineRunner
+from savant.utils.logging import get_logger
+from savant.utils.platform import is_aarch64
+
+
+def check_encoder_is_available(parameters: Dict[str, Any]) -> bool:
+    """Check if encoder is available."""
+
+    logger = get_logger(__name__)
+
+    output_frame = parameters.get('output_frame')
+    if not output_frame:
+        return True
+
+    codec = CODEC_BY_NAME[output_frame['codec']]
+    if codec not in [Codec.H264, Codec.HEVC]:
+        return True
+
+    logger.info('Checking if encoder for codec %r is available', output_frame['codec'])
+    pipeline: Gst.Pipeline = Gst.Pipeline.new()
+
+    converter_props = {}
+    if not is_aarch64():
+        converter_props['nvbuf-memory-type'] = int(pyds.NVBUF_MEM_CUDA_UNIFIED)
+
+    elements = [
+        PipelineElement(
+            'videotestsrc',
+            properties={'num-buffers': 50},
```

Comment:
Do we need 50 buffers here? Maybe 1 buffer is enough?

Reply:
Yes, 1 buffer would be enough.